org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/GradleBuild.java

```diff
@@ -62,14 +62,37 @@
 
     /**
      * Executes an action in the Gradle runtime.
+     *
+     * <p>
+     * This method instantiates a new connection to the Tooling API, pre-configures it with IDE
+     * services and executes the target action. Clients can use this method to load models and
+     * execute tasks, tests and custom build actions without explicitly configuring
+     * inputs/outputs/cancellation/etc.
+     *
      * <p>
-     * TODO (donat) explain how the action is executed (refer to Tooling API documentation)
+     * The following sections will show examples how can this API be used.
      * TODO (donat) explain common use-cases: task execution, test execution, and model loading
-     * TODO (donat) explain how the project connection is preconfigured: outputs and outputs to IDE console, progress and cancellation are wired to the monitor object
-     * TODO (donat) document exception handling
-     * TODO (donat) document scheduling rules
      * TODO (donat) document how to load custom models (here or link to external documentation)
      *
+     * <p>
+     * The following {@link ProjectConnection} methods that can't be wired into the Eclipse
+     * services. For that reason the target action cannot use them, otherwise an
+     * {@link UnsupportedOperationException} is thrown.
+     * <ul>
+     * <li>{@link ProjectConnection#model(Class)}</li>
+     * <li>{@link ProjectConnection#getModel(Class, org.gradle.tooling.ResultHandler))}</li>
+     * <li>{@link ProjectConnection#action()}</li>
```

Comment:
❓ Should these be deprecated because they have alternatives that are safe to use?

Reply:
❌ I see no reason why these methods wouldn't work. We should support all of them.

Reply:
Fixed